lint/lint.py

```diff
@@ -613,50 +634,76 @@ def output_errors_text(errors):
         pos_string = path
         if line_number:
             pos_string += ":%s" % line_number
-        print("%s: %s (%s)" % (pos_string, description, error_type))
+        logger.error("%s: %s (%s)" % (pos_string, description, error_type))
+
+def output_errors_markdown(errors):
+    if not errors:
+        return
+    heading = """Got lint errors:
+
+| Error Type | Position | Message |
+|------------|----------|---------|"""
+    for line in heading.split("\n"):
+        logger.error(line)
+    for error_type, description, path, line_number in errors:
+        pos_string = path
+        if line_number:
+            pos_string += ":%s" % line_number
+        logger.error("%s | %s | %s |" % (error_type, pos_string, description))
 
 def output_errors_json(errors):
     for error_type, error, path, line_number in errors:
         print(json.dumps({"path": path, "lineno": line_number,
                           "rule": error_type, "message": error}))
 
+
 def output_error_count(error_count):
     if not error_count:
         return
 
     by_type = " ".join("%s: %d" % item for item in error_count.items())
     count = sum(error_count.values())
+    logger.info("")
     if count == 1:
-        print("There was 1 error (%s)" % (by_type,))
+        logger.info("There was 1 error (%s)" % (by_type,))
     else:
-        print("There were %d errors (%s)" % (count, by_type))
+        logger.info("There were %d errors (%s)" % (count, by_type))
 
 def parse_args():
     parser = argparse.ArgumentParser()
     parser.add_argument("paths", nargs="*",
                         help="List of paths to lint")
     parser.add_argument("--json", action="store_true",
                         help="Output machine-readable JSON format")
+    parser.add_argument("--markdown", action="store_true",
+                        help="Output markdown")
     parser.add_argument("--css-mode", action="store_true",
                         help="Run CSS testsuite specific lints")
     return parser.parse_args()
 
 def main(force_css_mode=False):
     args = parse_args()
-    paths = list(args.paths if args.paths else all_filesystem_paths(repo_root))
-    return lint(repo_root, paths, args.json, force_css_mode or args.css_mode)
-
-def lint(repo_root, paths, output_json, css_mode):
+    if args.json and args.markdown:
+        logger.critical("Cannot specify --json and --markdown")
+        sys.exit(2)
+    paths = args.paths if args.paths else all_filesystem_paths(repo_root)
```

Comment:
This still needs to be `paths = list(...)`, otherwise it will throw a `TypeError: 'generator' object has no attribute '__getitem__'` error.